traffic_ops/testing/api/v4/cdn_locks_test.go

```diff
@@ -16,470 +16,296 @@ package v4
+						"typeId":    -1,
+					},
+					expectations: utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusForbidden)),
+				},
+			},
 		}
-	}
-	return "", serverID
-}
 
-func getCDNDetailsAndTopologyName(t *testing.T) (int, string, string) {
-	opts := client.NewRequestOptions()
-	topologiesResp, _, err := TOSession.GetTopologies(client.RequestOptions{})
-	if err != nil {
-		t.Fatalf("couldn't get topologies, err: %v", err)
-	}
-	if len(topologiesResp.Response) == 0 {
-		t.Fatal("no topologies returned")
-	}
-	for _, top := range topologiesResp.Response {
-		for _, node := range top.Nodes {
-			opts.QueryParameters.Set("name", node.Cachegroup)
-			cacheGroupResp, _, err := TOSession.GetCacheGroups(opts)
-			if err != nil {
-				t.Errorf("error while GETting cachegroups: %v", err)
-			}
-			if len(cacheGroupResp.Response) != 0 && cacheGroupResp.Response[0].ID != nil {
-				cacheGroupID := *cacheGroupResp.Response[0].ID
-				opts.QueryParameters.Del("name")
-				opts.QueryParameters.Set("cachegroup", strconv.Itoa(cacheGroupID))
-				serversResp, _, err := TOSession.GetServers(opts)
-				if err != nil {
-					t.Errorf("couldn't get servers: %v", err)
+		for method, testCases := range methodTests {
+			t.Run(method, func(t *testing.T) {
+				for name, testCase := range testCases {
+
+					topology := ""
+					cdnLock := tc.CDNLock{}
+					cacheGroup := tc.CacheGroupNullable{}
+					topQueueUp := tc.TopologiesQueueUpdateRequest{}
+
+					if testCase.requestOpts.QueryParameters.Has("topology") {
+						topology = testCase.requestOpts.QueryParameters.Get("topology")
+					}
+
+					if testCase.requestBody != nil {
+						if getId, ok := testCase.requestBody["cdnId"]; ok {
+							testCase.requestBody["cdnId"] = getId.(func() int)()
+							dat, err := json.Marshal(testCase.requestBody)
+							assert.NoError(t, err, "Error occurred when marshalling request body: %v", err)
+							err = json.Unmarshal(dat, &topQueueUp)
+							assert.NoError(t, err, "Error occurred when unmarshalling request body: %v", err)
+						} else if typeName, ok := testCase.requestBody["typeName"]; ok {
+							testCase.requestBody["typeId"] = GetTypeId(t, typeName.(string))
+							dat, err := json.Marshal(testCase.requestBody)
+							assert.NoError(t, err, "Error occurred when marshalling request body: %v", err)
+							err = json.Unmarshal(dat, &cacheGroup)
+							assert.NoError(t, err, "Error occurred when unmarshalling request body: %v", err)
+						} else {
+							dat, err := json.Marshal(testCase.requestBody)
+							assert.NoError(t, err, "Error occurred when marshalling request body: %v", err)
+							err = json.Unmarshal(dat, &cdnLock)
+							assert.NoError(t, err, "Error occurred when unmarshalling request body: %v", err)
+						}
+					}
+
+					switch method {
+					case "GET":
+						t.Run(name, func(t *testing.T) {
+							resp, reqInf, err := testCase.clientSession.GetCDNLocks(testCase.requestOpts)
+							for _, check := range testCase.expectations {
```

Comment:
In the[ old code base](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/testing/api/v4/cdn_locks_test.go#L141), we were validating fields after we called CDNLocks. Should we follow the same and call validateResponseFields() just like we do for POST case?